### ISSUE TITLE
Cannot get sourcemaps to work for uploaded files

### ISSUE THREAD

**Issue by gabaum10** [bugsnag/webpack-bugsnag-plugins#5]
I have no idea where to post this and I do apologize if this is not the place, but the docs are incredibly unclear as to which options you're supposed to use for what for the source map uploads.  I'm trying to add the source map upload plugin to a [webpack template ](https://github.com/vuetifyjs/webpack)

Basically, I have a setup running in dev behind an internal firewall.  I've integrated the webpack sourcemap upload plugin and configured it like so:

```
new BugsnagSourceMapUploaderPlugin({
      apiKey: '*****',
      appVersion: appVersion,
      uploadSource: true
})
```

This correctly uploads them and I can see them in he bugsnag console.  However, when an error comes in it does not match.  The app versions are all correctly aligned.  Now, the public path is configured relatively based on an index.html file like this:

```
    // Template for index.html
    index: path.resolve(__dirname, '../dist/index.html'),

    // Paths
    assetsRoot: path.resolve(__dirname, '../dist'),
    assetsSubDirectory: 'static',
    assetsPublicPath: '/',

    /**
     * Source Maps
     */

    productionSourceMap: true,
    // https://webpack.js.org/configuration/devtool/#production
    devtool: '#source-map',
```

Great.  The sourcemaps are uploaded and look like this in the console: `/static/js/vendor.3f9cb3774b75d70c28a4.js`

What configuration option am I missing here?  Is this a matter of the original source files living behind the firewall?  I thought that was the whole point of uploading the files?  If this isn't the place to ask this, please direct me where I should go to try and figure this out.

**Comment by gabaum10** [comment]
Interestingly, this may be related to #3?

**Comment by bengourley** [comment]
Hi @gabaum10. Here's good for issues – equally you can email support@bugsnag.com so that we can tie in your report with your Bugsnag account and get a bit more context.

To resolve this issue, you should be able to use the [wildcard functionality](https://docs.bugsnag.com/api/js-source-map-upload/#do-you-support-partial-matching-wildcards-for-the-minified-url) in the `publicPath`/`minifiedUrl`options…

> The asterisk character can be used as a wildcard for the `minifiedUrl` field and source file field names.
> 
> Multiple wildcards can be used in the same field. Each asterisk matches zero or more characters.


_Note: `publicPath`/`minifiedUrl` refer to the same thing, but the HTTP API uses `minifiedUrl` and the JS source map tool uses `publicPath`. This is confusing, sorry!_

I can't see exactly from the details that you posted, but you should be able to set `publicPath` to something like the following in order to match your asset urls:

```
new BugsnagSourceMapUploaderPlugin({
      apiKey: '*****',
      appVersion: appVersion,
      uploadSource: true,
      publicPath: 'http*://your.domain.com/static/js/vendor.*.js'
})
```

_Note: you should tweak this to your needs – the `http*` part is what you would need to do if you're serving the js on both http and https. The `vendor.*.js` should cover you for the different webpack hashes._

In summary, you basically want the `publicPath` option to match the url that the browser will load the JS from, which is what I think you were getting at when you said this:

> The sourcemaps are uploaded and look like this in the console: /static/js/vendor.3f9cb3774b75d70c28a4.js

Let me know if this helps. Cheers!

**Comment by gabaum10** [comment]
@bengourley so can you do something like:

`https://your.domain.com/static/js/*.js` to catch any of the JS files?

**Comment by gabaum10** [comment]
When I tried to set it to that, the uploaded source maps have a url of `https://dev.domain.com/static/js/*.js/static/js/vendor.3f9cb3774b75d70c28a4.js` which is obviously not right.  Does the plugin append the public path that's passed with the public path in the config?

**Comment by gabaum10** [comment]
Ah!  I got it working by using this: `publicPath: '*/'`

**Comment by bengourley** [comment]
> Does the plugin append the public path that's passed with the public path in the config?

Yes, you are right – the absolute value I provided would work with the underlying [`bugsnag-sourcemaps`](https://github.com/bugsnag/bugsnag-sourcemaps) module but as webpack can create multiple maps/bundles in one build, this option is the public path (web address) to the static directory.

> Ah! I got it working by using this: publicPath: '*/'

Glad you got this working! As long as each time you generate a bundle/hash you upload the sourcemaps, you should be good with this setting 👍

**Comment by gabaum10** [comment]
Thanks for the help!

**Comment by konstantin24121** [comment]
> Ah! I got it working by using this: publicPath: '*/'

Not working for me(

I try load my source map with same options, but bugsnag does not see it in dashboard 

```javascript
new BugsnagSourceMapUploaderPlugin({
      apiKey: global.bugsnagId,
      appVersion: global.appMeta.version,
      overwrite: true,
      publicPath: '*/',
    }),
```
![image](https://user-images.githubusercontent.com/9361325/47318810-bdd1af00-d655-11e8-8541-7cec907adeaa.png)

![image](https://user-images.githubusercontent.com/9361325/47318725-7ea35e00-d655-11e8-93dd-9811afd218ae.png)

**Comment by snmaynard** [comment]
Source maps are only applied going forwards - so when you upload a source map you will have to trigger another error to verify its working. Does a subsequent error use the source maps properly?

**Comment by konstantin24121** [comment]
@snmaynard thank you! All work fine guys
